src/v/v8_engine/v8_batch_consumer.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "kafka/protocol/batch_consumer.h"
+#include "model/record.h"
+#include "v8_engine/executor.h"
+#include "v8_engine/scripts_dispatcher.h"
+#include <seastar/core/loop.hh>
+
+namespace v8_engine {
+
+template<typename Executor>
+class v8_batch_consumer {
+public:
+    explicit v8_batch_consumer(script_with_env_wrapper<Executor> scripts_with_env)
+    : _scripts_with_env(scripts_with_env) {};
+
+    v8_batch_consumer(const v8_batch_consumer& o) = delete;
+    v8_batch_consumer& operator=(const v8_batch_consumer& o) = delete;
+    v8_batch_consumer& operator=(v8_batch_consumer&& o) noexcept = default;
+    v8_batch_consumer(v8_batch_consumer&& o) noexcept = default;
+    ~v8_batch_consumer() = default;
+
+    ss::future<ss::stop_iteration> operator()(model::record_batch batch) {
+        co_await _scripts_with_env.script->run(batch, _scripts_with_env.executor);
```

Comment:
Catch possible exceptions here and stop the iteration:
```
try {
  logic...
} catch(ex) {
  _ex = ex;
  co_return ss::stop_iteration::yes;
}
```

Reply:
Fix it, added try-catch in fetch.cc